src/trackingModules/CanvasTracking.js

```diff
@@ -0,0 +1,53 @@
+export const CANVAS_EVENTS = {
+  '.djs-context-pad .entry': {
+    eventName: 'canvas.contextPad.click',
+    data: 'data-action'
+  },
+  '.djs-popup .entry': {
+    eventName: 'canvas.popup-menu.click',
+    data: 'data-id'
+  },
+  '.djs-palette-entries .entry': {
+    eventName: 'canvas.palette.click',
+    data: 'data-action'
+  }
+};
+
+export default class CanvasTracking {
+  constructor(props) {
+
+    this._selection = props._selection;
+    this.sendEvent = props.track;
+  }
+
+  trackEvent(event) {
+    let payload = {};
+    const element = event.target;
+
+    Object.keys(CANVAS_EVENTS).forEach(eventSelector => {
+      if (matches(element, eventSelector)) {
+        const canvasEvent = CANVAS_EVENTS[eventSelector];
+
+        payload[`${canvasEvent.data}`] = element.getAttribute(canvasEvent.data);
+        payload['selection'] = this._selection.get();
+
+        this.sendEvent(canvasEvent.eventName, payload);
```

Comment:
Why not emit this as an eventBus event? Anyone can hook into it to add additional data / context.

Reply:
Externally the core tracking extension could still translate this to calls to some API method (you call it `bpmnjstracking#track).

Reply:
This is just what we do internally and it is powerful (i.e. you can plug-in and add your payload). But happy to keep it plain simple in the first version.